### PR TITLE
Port axum-extra

### PR DESCRIPTION
**Pull request by davidpdrsn**
This ports the code and docs in axum-extra. `cargo t --all-features -p axum-extra` now passes. Getting closer 😊

### CODE REVIEW

File: axum-extra/src/either.rs [hunk 1]
```diff
@@ -184,17 +219,22 @@ pub enum Either8<E1, E2, E3, E4, E5, E6, E7, E8> {
...
+
 macro_rules! impl_traits_for_either {
     (
         $either:ident =>
         [$($ident:ident),* $(,)?],
         $last:ident $(,)?
     ) => {
         #[async_trait]
-        impl<S, B, $($ident),*, $last> FromRequest<S, B> for $either<$($ident),*, $last>
+        impl<S, B, $($ident),*, $last> FromRequest<S, B, private::EitherOnce> for $either<$($ident),*, $last>
```
Comment (davidpdrsn) [inline axum-extra/src/either.rs:234]: This required the same trick as the tuples.

File: axum-core/Cargo.toml [hunk 2]
```diff
@@ -9,7 +9,7 @@ license = "MIT"
 name = "axum-core"
 readme = "README.md"
 repository = "https://github.com/tokio-rs/axum"
-version = "0.2.6" # remember to also bump the version that axum depends on
+version = "0.2.6" # remember to also bump the version that axum and axum-extra depends on
```
Comment (jplatte) [inline axum-core/Cargo.toml:12]:
```suggestion
version = "0.2.6" # remember to also bump the version that axum and axum-extra depend on
```

File: axum-extra/src/either.rs [hunk 3]
```diff
@@ -184,17 +219,22 @@ pub enum Either8<E1, E2, E3, E4, E5, E6, E7, E8> {
...
         where
             $($ident: FromRequestParts<S>),*,
-            $last: FromRequest<S, B>,
+            $last: FromRequest<S, B, axum_core::extract::private::Once>,
```
Comment (jplatte) [inline axum-extra/src/either.rs:237]: I wonder how useful this impl is at all, with just the last of the alternatives being allowed to take the request. Maybe we should only provide `FromRequestParts`? We would also need another impl with `TupleOnce` instead `Once` if we wanted to have a request-consuming tuples as `Either`s last type parameter.

Comment (davidpdrsn) [reply axum-extra/src/either.rs]: Yeah I agree. I've removed it.

**Review by jplatte — APPROVED** [review]

**Pull request merged**